### PR TITLE
PLAT-2502: Configurable Subnet Sizes

### PR DESCRIPTION
**Pull request by Secretions**
The `cidr_mask` variable for the public/private subnets has been hardcoded. This makes it configurable.

### CODE REVIEW

File: cdk/domino_cdk/config/template.py [hunk 1]
```diff
@@ -78,6 +78,8 @@ def add_nodegroups(name, count, min_size, instance_types, labels, disk_size=100,
         id=None,
         create=True,
         cidr='10.0.0.0/16',
+        public_cidr_mask=27,
```
Comment (ddl-denis-parnovskiy) [inline cdk/domino_cdk/config/template.py:81]: In 208.128.0.0/**11** and **255.224.0.0** "11" is called prefix and "255.224.0.0" is called mask.

File: cdk/domino_cdk/config/vpc.py [hunk 2]
```diff
@@ -11,6 +11,10 @@ class VPC:
     id: vpc-abc123 - VPC id when using an existing VPC
     cidr: 10.0.0.0/16 - Primary CIDR range for VPC
                         NOTE: EKS needs _lots_ of IPs
+    private_cidr_mask: 19 - Size of private subnets. Should be large enough
```
Comment (ddl-denis-parnovskiy) [inline cdk/domino_cdk/config/vpc.py:14]: I'm confused. Is 19 a prefix, or a mask or a network size? It looks like a prefix but is called mask on left side and size of subnet on the right size. The network size is reverse logarithmic proportional to the prefix length. I.e. prefix length 32 means network size of 1.

File: cdk/domino_cdk/config/template.py (hunk 1 above)
Comment (Secretions) [reply cdk/domino_cdk/config/template.py:81]: I believe `208.120.0.0` is the prefix, `255.224.0.0` us the subnet mask, and `11` is just the cidr notation version of 255.224.0.0 (ie you could read it as `208.128.0.0/255.224.0.0`--they mean the same thing).

Anyhow, the choice is naming is to reflect the api call to `SubnetConfiguration`, which uses `cidr_mask` to name its value (see vpc provisioner in this same PR). I don't believe the terminology here is inaccurate (subnet mask given as cidr value), and would prefer to keep the name mapping the same as the API.

File: cdk/domino_cdk/config/vpc.py (hunk 2 above)
Comment (Secretions) [reply cdk/domino_cdk/config/vpc.py]: Subnet size in CIDR notation. I updated the phrasing to specify that it's CIDR size.

**Review by steved — APPROVED** [review]

**Pull request merged**